tests/lit-tests/signed-integer-overflow.ispc

```diff
@@ -0,0 +1,186 @@
+
+unmasked uint16 add_u16(uint16 a, uint16 b) { return a + b; }
+
+unmasked uint8 add_u8(uint8 a, uint8 b) { return a + b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @sub_64
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @sub_32
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @sub_16
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @sub_8
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i8
+
+unmasked int64 sub_64(int64 a, int64 b) { return a - b; }
+
+unmasked int32 sub_32(int32 a, int32 b) { return a - b; }
+
+unmasked int16 sub_16(int16 a, int16 b) { return a - b; }
+
+unmasked int8 sub_8(int8 a, int8 b) { return a - b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @sub_u64
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @sub_u32
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @sub_u16
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @sub_u8
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i8
+
+unmasked uint64 sub_u64(uint64 a, uint64 b) { return a - b; }
+
+unmasked uint32 sub_u32(uint32 a, uint32 b) { return a - b; }
+
+unmasked uint16 sub_u16(uint16 a, uint16 b) { return a - b; }
+
+unmasked uint8 sub_u8(uint8 a, uint8 b) { return a - b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @shl_64
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @shl_32
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @shl_16
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @shl_8
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i8
+
+unmasked int64 shl_64(int64 a, int64 b) { return a << b; }
+
+unmasked int32 shl_32(int32 a, int32 b) { return a << b; }
+
+unmasked int16 shl_16(int16 a, int16 b) { return a << b; }
+
+unmasked int8 shl_8(int8 a, int8 b) { return a << b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @shl_u64
+// CHECK_NSW-NOT:             %{{.*}}= shl{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @shl_u32
+// CHECK_NSW-NOT:             %{{.*}}= shl{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @shl_u16
+// CHECK_NSW-NOT:             %{{.*}}= shl{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @shl_u8
+// CHECK_NSW-NOT:             %{{.*}}= shl{{.*}}nsw{{.*}}i8
+
+unmasked uint64 shl_u64(uint64 a, uint64 b) { return a << b; }
```

Comment:
As Aleksei mention, `nsw` is not expected on shift operations. And I suggest checking it the test that it's not set for operations that do not assume it.